glean-core/rlb/src/lib.rs

```diff
@@ -411,6 +422,13 @@ pub fn register_ping_type(ping: &private::PingType) {
                 glean.register_ping_type(&ping.ping_type);
             })
         })
+    } else {
+        // We need to keep track of pings, so they get re-registered after a reset.
+        // This state is kept across Glean resets, which should only ever happen in test mode.
+        // It's a set and keeping them around forever should not have much of an impact.
+        let m = PRE_INIT_PING_REGISTRATION.get_or_init(Default::default);
+        let mut lock = m.lock().unwrap();
+        (*lock).push(*ping);
```

Comment:
@badboy this really makes Rust unhappy, and I think I found a solution, but that got me into a different rabbit-hole, so I'm questioning my approach.

This line errors out because I'm trying to move `ping` (thanks borrow checker!).

If i store references in the global vector (i.e. `&private::PingType`), then this whole file gets happy. However, it requires me to require a `static` lifetime on the `register_ping_type` function, which in turns makes [this](https://searchfox.org/glean/rev/e9e1f0a83036cc62a63ea9adb34bff90f7a54a20/glean-core/rlb/src/private/ping.rs#38) unhappy. We know that, in practice, our generated code will always use `static` lifetime, so this change seems ok, but I can't find a way to communicate this to the compiler. Or is this whole approach wrong?

Reply:
What if we clone the ping? We can worry about making it more efficient later (and it's not like we register hundreds of custom pings).
(You also don't need the `(*lock)`, `lock` should suffice)

Reply:
Good point.